src/app/qgisapp.cpp

```diff
@@ -17581,3 +17602,47 @@ void QgisApp::showEvent( QShowEvent *event )
     }
   } );
 }
+
+void QgisApp::handleRenderedLayerStatistics( const QgsMapCanvas *mapCanvas ) const
+{
+  const QgsRenderedItemResults *renderedItemResults = mMapCanvas->renderedItemResults( false );
+  if ( !renderedItemResults )
+  {
+    return;
+  }
+
+  QList<const QgsRenderedLayerStatistics *> computedStatistics;
+  for ( const QgsRenderedItemDetails *item : renderedItemResults->renderedItems() )
+  {
+    if ( const QgsRenderedLayerStatistics *layerStatistics = dynamic_cast< const QgsRenderedLayerStatistics *>( item ) )
+    {
+      computedStatistics.append( layerStatistics );
+    }
+  }
+
+  if ( !computedStatistics.empty() )
+  {
+    for ( QgsMapLayer *layer : mMapCanvas->layers() )
```

Comment:
This should probably be modified to loop through the statistics, and then retrieve the layer from the project by the ID from the statistics accordingly.

Reply:
Indeed. Done.